### PR TITLE
Added Tests and CI

### PR DESCRIPTION
**Pull request by saltysealion**

### CODE REVIEW

File: tests/features/steps/steps.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        context.exception = None
+    except Exception as exception:
+        context.exception = exception
+    
+@then('a {object_type:w} object is returned')
+def step_impl(context, object_type):
+    assert context.result.type == object_type
+
+@then('it contains {variable_name:w}')
+@then('it contains a {variable_name:w}')
```
Comment (stabacco) [inline tests/features/steps/steps.py]: looks like this is a duplicate line

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          name: run tests
+          command: pipenv run behave ~/app/tests/features
+
+
+workflows:
+  version: 2
+  build_and_test:
+    jobs:
+      - build_and_test:
+          context: q-ctrl-api-development
```
[the remark this answers is not in the document]
Comment (saltysealion) [reply .circleci/config.yml]: Yes, it's required

File: tests/features/steps/steps.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+from behave import *
+from qctrl import Qctrl
+import ast
+import numpy as np
+import os
+
+KEY = os.environ.get('TEST_API_KEY')
```
[the remark this answers is not in the document]
Comment (saltysealion) [reply tests/features/steps/steps.py]: Ok agreed

File: tests/features/steps/steps.py (hunk 1 above)
Comment (saltysealion) [reply tests/features/steps/steps.py]: One line contains an additional letter, both steps are treated the same

**Pull request merged**